### PR TITLE
[HB-7028] Handle Memory Leak

### PR DESCRIPTION
**Pull request by cb-jpadilla**
- Fixes a memory leak for fullscreen ads.
- Updated CHANGELOG, README, Adapter version to 4.6.2.0.4

### CODE REVIEW

**Comment by cb-jpadilla** [comment]
This fixes the leak. Trying the approach in which we create a new class for the Listener causes the close delegate to no longer trigger. 🤔 May need to look more into it, if we want to go with adding more preventive measure.

**Comment by cb-jpadilla** [comment]
Will be pushing another commit that moves the listener to an internal class.

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 1]
```diff
@@ -831,4 +733,174 @@ class HyprMXAdapter : PartnerAdapter {
...
+                    it.resume(result)
+                }
+            } ?: run {
+                PartnerLogController.log(LOAD_FAILED, "Unable to resume continuation. Continuation is null.")
+            }
+        }
+
+        override fun onAdStarted(placement: Placement?) {
+            PartnerLogController.log(SHOW_SUCCEEDED)
+            onShowSuccess()
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:761]: let's clear the lambda after calling this.

File: HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt [hunk 2]
```diff
@@ -831,4 +733,174 @@ class HyprMXAdapter : PartnerAdapter {
...
```
Comment (bichenwang) [inline HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt:839]: let's clear the lambda after calling this

**Review by bichenwang — APPROVED** [review]
lgtm otherwise

**Review by bichenwang — APPROVED** [review]

**Pull request merged**